src/schemathesis/models.py

```diff
@@ -178,8 +179,12 @@ def call(
             close_session = False
 
         data = self.as_requests_kwargs(base_url)
+        if "headers" not in data or data["headers"] is None:
+            data.update({"headers": {}})
         if headers is not None:
-            data["headers"] = {**(data["headers"] or {}), **headers}
+            data["headers"].update(headers)
+        if "User-Agent" not in data["headers"]:
+            data["headers"].update({"User-Agent": USER_AGENT})
```

Comment:
Instead of using `update` with a single key, I'd prefer using `__setitem__` - `data["headers"]["User-Agent"] = USER_AGENT`. It is more readable and slightly faster